src/main/java/org/hydev/mcpm/client/installer/InstallInteractor.java

```diff
@@ -73,36 +82,59 @@ public boolean installPlugin(InstallInput installInput,
         var idPluginModel = pluginModel.id();
         var pluginVersion = pluginModel.getLatestPluginVersion().orElse(null);
         if (pluginVersion == null) {
-            installPresenter.displayResult(new InstallResult(Type.NO_VERSION_AVAILABLE), installInput.name());
-            return false;
+            return List.of(new InstallResult(Type.NO_VERSION_AVAILABLE, pluginName));
         }
 
-        // Name of the latest version plugin
-        String pluginName = pluginVersion.meta().name();
-
-        if (databaseManager.checkPluginInstalledByName(pluginName)) {
-            InstallResult installResult = new InstallResult(Type.PLUGIN_EXISTS);
-            installPresenter.displayResult(installResult, installInput.name());
+        var results = new ArrayList<InstallResult>();
```

Comment:
You can actually say `new ArrayList<>();` and Java will infer the generics for you :) just something to save you time for next time